CONTRIBUTING.md

```diff
@@ -16,15 +16,186 @@ GitBook (which we use for documentation) uses Markdown as its file format.
+
+Chances are, however, you don't want `_index.md` to be a list of that subdirectory's pages (especially if it's a new language), but rather that it's treated as a single page would: display content and contain a table of contents. You can tell Hugo to do this by including the `layout` attribute in its front matter.
+
+```
+---
+title: "New language"
+weight: 6
+layout: single
+---
+```
+
+`_index.md` will now use the layout for single pages (`layouts/_default/single.html`).
+
+#### Sidebar and pages
+
+If you would like the title of a given page to appear differently in the sidebar - say, a shorter title there, but a more descriptive title on the page). You can use the `sidebarTitle` attribute in the front matter to define it, while the regular `title` attribute will only be used on the page itself.
+
+```
+---
+title: "New language support details"    # Title on page
+sidebarTitle: "New language"             # How title appears in sidebar
+weight: 6
+layout: single
+---
+```
+
+#### Disabling table of contents
+
+By default, all pages that use the `single` layout will generate a table of contents beneath their title. If you would like to disable it, add `toc` attribute to your front matter:
+
+```
+---
+title: "New language"
+weight: 4
+toc: false
+---
+```
+
+### Shortcodes
+
+There are a few shortcodes written for the documentation that keep styling consistent and insert specific HTML into pages where they are referenced.
+
+#### Notes
+
+For the moment, Hugo does not provide a way to pass values to Markdown's blockquote syntax, but in this documentation there are a few variations on the blockquote that are used for different purposes. So, when leaving a "Note" or blockquote on your pages, use the `note` shortcode instead
+
+```
+{{< note >}}
+Lorem ipsum
+{{< /note }}
+```
+
+This shortcode will automatically create the **Note:** first line for you.
+
+`note` accepts two optional parameters:
+
+- `theme`: default none, standard note. (`info`, `warning`). This will change the color and the default first line.
+- `title`: default dependent on theme. Over-rides that default.
+
+```
+{{< note theme="warning" title="Danger, Will Robinson" >}}
+Lorem ipsum!
+{{< /note >}}
+```
+
+#### Images
+
+The standard Markdown syntax for images remains the same, but it is possible to pass values to it that modify the appearance of the image
+
+```
+<!-- Normal image syntax -->
+![alt-title](/images/current_image.png)
+
+<!-- Scaled image -->
+![alt-title](/images/current_image.png "0.5")
+
+<!-- Scaled image displayed inline, such as an icon in a sentence -->
+![alt-title](/images/current_image.png "0.1-inline")
+```
+
+#### Videos & asciinema
+
+Displaying videos and asciinema recordings requires their own shortcodes, but they have identical parameters:
```

Comment:
Are there parameters beyond src?  (HTML video has a whole crapload of other options.)

Reply:
At the moment no. What would you  want to  add?

Reply:
At minimum I think `type` is required, but there's a whole lot more: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video

Ideally we'd just allow these to pass-through rather than explicitly supporting each of them?  (Only applies to video, not asciinema, which has a different set of possible attributes.)

Reply:
Ya right now it's just set firm at `type="video/mp4"`. I'd say we set a few reasonable defaults and then accept everything to be overwritten.